scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,172 @@
+"""Module to generate a column map to their corresponding statistical variable defintions."""
+
+import os
+import json
+
+import pandas as pd
+
+#TODO:Remove this import and make it from ~/data/utils/ directory when it is available
+os.system(
+    'curl https://raw.githubusercontent.com/Abilityguy/data/dcid_gen/util/statvar_dcid_generator.py -LO statvar_dcid_generator.py'
+)
+from statvar_dcid_generator import get_stat_var_dcid
+
+
+def generate_stat_var_map(spec_dict: dict, column_list: list) -> dict:
+    """Wrapper function for generateColMapBase class to generate column map.
+
+  Args:
+    specDict:
+      A dictionary containing specifications for the different properties of the statistical variable.
+    columnList:
+      A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+
+  Returns:
+    A dict mapping each column to their respective stat_var node definitions.
+    Example output: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }
+  """
+    col_map_obj = GenerateColMapBase(spec_dict=spec_dict,
+                                     column_list=column_list)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+    delimiter: The delimiting string that is used for tokenising the column name
+  """
+
+    def __init__(self, spec_dict={}, column_list=[], delimiter='!!'):
```

Comment:
maybe useful to take a dictionary of counters.

This would be useful to show summary stats of number of columns, number of statvars generated, statvars of differnt population types, errors or warnings,

Reply:
Why was this comment resolved? Please only resolve after responding / implementing the suggested change. In either case, it's nice to acknowledge the comment!

Reply:
Marked with [TODO](https://github.com/sharadshriram/data/blob/acs_common/scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py#L155-L156) in the code to revisit in the next PR.